btk/catalog.py

```diff
@@ -70,33 +70,62 @@ def from_file(cls, catalog_file, verbose=False):
     def _prepare_table(self):
         table = deepcopy(self._raw_catalog)
 
+
         # TODO: does the WLDCatalog require the 'ra' and 'dec' columns
         # convert ra dec from degrees to arcsec in catalog.
-        if "ra" in table.columns:
-            table["ra"] *= 3600
-        if "dec" in table.columns:
+        if "ra" in table:
+            table["Ra"] *= 3600
+        if "dec" in table:
             table["dec"] *= 3600
 
+        f = self._raw_catalog["fluxnorm_bulge"] / (
+                self._raw_catalog["fluxnorm_disk"] + self._raw_catalog["fluxnorm_bulge"]
+        )
+        r_sec = np.hypot(
+            self._raw_catalog["a_d"] * (1 - f) ** 0.5 * 4.66, self._raw_catalog["a_b"] * f ** 0.5 * 1.46
+        )
+        # BTK now requires ref_mags, but WLD still wants magnitudes
+        table["ref_mag"] = self._raw_catalog["i_ab"]
+        table["btk_size"] = r_sec
+        #ADds the extra columns to both catalogs just to be sure
+        self._raw_catalog["ref_mag"] = self._raw_catalog["i_ab"]
+        self._raw_catalog["btk_size"] = r_sec
+
         return table
 
 
 class CosmosCatalog(Catalog):
     @classmethod
-    def from_file(cls, catalog_file, verbose=False):
+    def from_file(cls, catalog_files, verbose=False):
+        """
+
+        Paramters
+        ---------
+        catalog_files: list of galsim cataolgs
+        """
         # This will return a COSMOSCatalog object.
-        catalog = galsim.COSMOSCatalog(file_name=catalog_file)
+        catalog_coord = astropy.table.Table.read(catalog_files[0], 'fits')
+        catalog_fit = astropy.table.Table.read(catalog_files[1], 'fits')
+        catalog = astropy.table.hstack([catalog_coord, catalog_fit])
```

Comment:
I'm no fan of passing lists as arguments, especially if the order in the list matters. 
Very hard to remember which order you should put them.

Do we need the two different catalogs ? If so we should use two distinct arguments.

Reply:
Sadely for galsim yes, there are two catalogs that contain slighlty different informations. Though we were thinking of sort of dropping the galsim catalog. It is not actually necessary at this point and we could use the default catalog to draw coordinates and fluxes as it was done before. Galsim cats only matter for the draw single part.

Reply:
But my opinion of catalog is shifting, and I actually don't think that we need to build a CosmosCatalog here, at least, not unless we want to draw HST images. If we want to generate Rubin images, we should rather use the WLD catalog.

Reply:
You mean that for Rubin images we should preferably use WLD only, or is it about the Catalog objects ?

Reply:
To generate Rubin images, you actually want a Rubin-compatible catalog with the magnitudes in the Rubin bands, that's what matters.

Reply:
I'll make an other PR for that it's a small rework.